cmd/neofs-adm/internal/modules/morph/verified_domains.go

```diff
@@ -0,0 +1,219 @@
+				if strPublicKeys[i] == strPublicKeys[j] {
+					return fmt.Errorf("duplicated public key %s", strPublicKeys[i])
+				}
+			}
+
+			buf := io.NewBufBinWriter()
+			emit.CheckSig(buf.BinWriter, bKey)
+
+			additionalRecords[i] = address.Uint160ToString(hash.Hash160(buf.Bytes()))
+		}
+	}
+
+	walletDir := config.ResolveHomePath(vpr.GetString(alphabetWalletsFlag))
+
+	wallets, err := openAlphabetWallets(vpr, walletDir)
+	if err != nil {
+		return err
+	}
+
+	committeeAcc, err := getWalletAccount(wallets[0], committeeAccountName)
+	if err != nil {
+		return fmt.Errorf("get committee account: %w", err)
+	}
+
+	n3Client, err := getN3Client(vpr)
+	if err != nil {
+		return fmt.Errorf("open connection: %w", err)
+	}
+
+	nnsContractAddr, err := nnsrpc.InferHash(n3Client)
+	if err != nil {
+		return fmt.Errorf("get NeoFS NNS contract address: %w", err)
+	}
+
+	actr, err := actor.NewSimple(n3Client, committeeAcc)
+	if err != nil {
+		return fmt.Errorf("init committee actor: %w", err)
+	}
+
+	domain := vpr.GetString(domainFlag)
+	scriptBuilder := smartcontract.NewBuilder()
+	hasOtherRecord := false
+	mAlreadySetIndices := make(map[int]struct{}, len(additionalRecords))
+
+	err = iterateNNSDomainTextRecords(actr, nnsContractAddr, domain, func(rec string) error {
+		for i := range additionalRecords {
+			if additionalRecords[i] == rec {
+				mAlreadySetIndices[i] = struct{}{}
+				return nil
+			}
+		}
+
+		hasOtherRecord = true
+
+		return errBreakIterator
+	})
+	if err != nil {
+		switch {
+		default:
+			return fmt.Errorf("handle domain %q records: %w", domain, err)
+		case errors.Is(err, errMissingDomainRecords):
+		case errors.Is(err, errDomainNotFound):
+			domainToRegister := domain
+			if labels := strings.Split(domainToRegister, "."); len(labels) > 2 {
+				// we need explicitly register L2 domain like 'some-org.neofs'
+				// and then just add records to inferior domains
+				domainToRegister = labels[len(labels)-2] + "." + labels[len(labels)-1]
+			}
+
+			scriptBuilder.InvokeMethod(nnsContractAddr, "register",
+				domainToRegister, committeeAcc.ScriptHash(), "ops@nspcc.ru", int64(3600), int64(600), int64(defaultExpirationTime), int64(3600))
+		}
+	}
+
+	if !hasOtherRecord && len(mAlreadySetIndices) == len(additionalRecords) {
+		cmd.Println("Current list is already the same, skip.")
+		return nil
+	}
+
+	if hasOtherRecord {
+		// there is no way to delete particular record, so clean all first
+		scriptBuilder.InvokeMethod(nnsContractAddr, "deleteRecords",
+			domain, nnsrpc.TXT.Int64())
+	}
+
+	for i := range additionalRecords {
+		if !hasOtherRecord {
+			if _, ok := mAlreadySetIndices[i]; ok {
+				continue
+			}
+		}
+
+		scriptBuilder.InvokeMethod(nnsContractAddr, "addRecord",
+			domain, nnsrpc.TXT.Int64(), additionalRecords[i])
+	}
+
+	txScript, err := scriptBuilder.Script()
+	if err != nil {
+		return fmt.Errorf("build transaction script: %w", err)
+	}
+
+	txID, vub, err := actr.SendRun(txScript)
```

Comment:
Why not `actr.Wait(actr.SendRun(txScript))`?

Reply:
to stay closer to other commands

Reply:
done